docs/source/plotting/MeshPlotHelp.rst

```diff
@@ -76,7 +81,294 @@ Basic example of plotting a general Poly3DCollection `Polygon <https://matplotli
+
+   sample = ws.sample()
+   environment = sample.getEnvironment()
+
+   mesh = sample.getShape().getMesh()
+   container_mesh = environment.getContainer().getShape().getMesh()
+
+   mesh_polygon_a = Poly3DCollection(mesh, facecolors = 'green', edgecolors='blue',alpha = 0.5, linewidths=0.1, zorder = 0.3)
+   mesh_polygon_b = Poly3DCollection(container_mesh, edgecolors='red', alpha = 0.1, linewidths=0.05, zorder = 0.5)
+   mesh_polygon_b.set_facecolor((1,0,0,0.5))
+
+   fig, axes = plt.subplots(subplot_kw={'projection':'mantid3d'})
+   axes.add_collection3d(mesh_polygon_a)
+   axes.add_collection3d(mesh_polygon_b)
+
+   for i in (1,3,5):
+      print(i)
+      mesh_polygon_i = Poly3DCollection(environment.getComponent(i).getMesh(), edgecolors='red', alpha = 0.1, linewidths=0.05, zorder = 0.5)
+      mesh_polygon_i.set_facecolor((1,0,0,0.5))
+      axes.add_collection3d(mesh_polygon_i)
+
+   # Auto scale to the mesh size
+   axes_lims = (-0.03,0.03)
+   axes.auto_scale_xyz(axes_lims, axes_lims, axes_lims)
+
+   axes.set_title('Pearl Sample in Container and Components(1,3,5)')
+   axes.set_xlabel('X / m')
+   axes.set_ylabel('Y / m')
+   axes.set_zlabel('Z / m')
+   axes.view_init(elev=5, azim=40)
+
+   def arrow(ax, vector, origin = None, factor = None, color = 'black',linestyle = '-'):
+      if origin == None:
+         origin = (ax.get_xlim3d()[1],ax.get_ylim3d()[1],ax.get_zlim3d()[1])
+      if factor == None:
+         lims = ax.get_xlim3d()
+         factor = (lims[1]-lims[0]) / 3.0
+      vector_norm = vector / np.linalg.norm(vector)
+      ax.quiver(
+            origin[0], origin[1], origin[2],
+            vector_norm[0]*factor, vector_norm[1]*factor, vector_norm[2]*factor,
+            color = color,
+            linestyle = linestyle
+      )
+   # Add arrow along beam direction
+   source = ws.getInstrument().getSource().getPos()
+   sample = ws.getInstrument().getSample().getPos() - source
+   arrow(axes, sample, origin=(0,0,-0.04))
+
+   plt.show()
+
+
+Add Arrows for Beam or Crystal Axes
+###################################
+
+In the above Containers example, a black arrow for the beam direction was added. Below, the real and reciprocal lattice
+vectors have been plotted (in solid and dashed linestyles respectively). Both of them make use of the ``arrow()`` function here:
+
+.. code-block:: python
+
+   def arrow(ax, vector, origin = None, factor = None, color = 'black',linestyle = '-'):
+    if origin == None:
+        origin = (ax.get_xlim3d()[1],ax.get_ylim3d()[1],ax.get_zlim3d()[1])
+    if factor == None:
+        lims = ax.get_xlim3d()
+        factor = (lims[1]-lims[0]) / 3.0
+    vector_norm = vector / np.linalg.norm(vector)
+    ax.quiver(
+         origin[0], origin[1], origin[2],
+         vector_norm[0]*factor, vector_norm[1]*factor, vector_norm[2]*factor,
+         color = color,
+         linestyle = linestyle
+    )
+
+   '''Add arrow along beam direction'''
+   source = ws.getInstrument().getSource().getPos()
```

Comment:
Similar comment to higher up. At the moment this script doesn't have a line to create ws. I'm guessing this is intentional to avoid repeating the python to do this but perhaps a single line comment saying "create ws as previously described" would be useful